### PR TITLE
Recipemd

### PR DESCRIPTION
**Pull request by careilly**
Added the ability to parse RecipeMD files, with a preference to choose which format to use.

### CODE REVIEW

**Comment by tmayoff** [comment]
Could you remove all the .swp files

**Comment by tmayoff** [comment]
I've been wondering if it's possible to support both recipemd and not, since the main difference between the two is the ingredients are surrounded by '---', could we for now instead check to see if the third (and possibly fourth) '---' lines exist and parse the ingredients that way. So we could have notes that are parsed both ways in the same vault, without a global switch

There's also checking for whether note being parsed currently has frontmatter in it or not. The file metadata from obsidian's API might have the information on it

No worries if you don't / can't contribute much more we can merge this with the few changes and add the rest of the above as issues for me another time

**Comment by careilly** [comment]
It does currently check for metadata, though it's doing it directly by inspecting the content rather than using the API as it probably should.

**Comment by careilly** [comment]
No idea why this closed.

File: src/recipe/ingredients.ts [hunk 1]
```diff
@@ -49,8 +56,48 @@ function parse_ingredients(content: string): Ingredient[] {
     return recipes;
 }
 
+function parse_ingredients_recipemd(content: string): Ingredient[] {
+    const debug = false;
```
Comment (tmayoff) [inline src/recipe/ingredients.ts:60]: If this is just going to be turned off here I'd rather just remove the debug logging completely here and setup a setting for it.

File: src/recipe/ingredients.ts [hunk 2]
```diff
@@ -49,8 +56,48 @@ function parse_ingredients(content: string): Ingredient[] {
...
+    
+
+    if (content.substring(0,2)=='---') {
+      ingredients = content.split('---')[3];
+    } else {
+      ingredients = content.split('---')[1];
+    }
+
+    if (debug) {
+      console.log("INGREDIENTS", ingredients);
```
Comment (tmayoff) [inline src/recipe/ingredients.ts:72]: If the log statements are going to be kept, could you add the function name to the beginning of the message

File: src/recipe/ingredients.ts [hunk 3]
```diff
@@ -49,8 +56,48 @@ function parse_ingredients(content: string): Ingredient[] {
...
+    }
+
+
+    if (typeof ingredients == 'undefined' || ingredients.length<=0) {
+        return new Array();
+    }
+    if (debug) {
+      console.log("PROCESSING", ingredients);
```
Comment (tmayoff) [inline src/recipe/ingredients.ts:80]: Same here



If the log statements are going to be kept, could you add the function name to the beginning of the message

File: test_vault/.obsidian/plugins/obsidian-meals/manifest.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "tmayoff-meals",
     "name": "Meal Plan",
-    "version": "0.0.8",
+    "version": "0.0.9",
```
Comment (tmayoff) [inline test_vault/.obsidian/plugins/obsidian-meals/manifest.json:4]: undo this, I don't want to make a release with this PR immediatly

**Review by tmayoff — CHANGES_REQUESTED** [review]

**Comment by tmayoff** [comment]
Also do you mind merging the latest of main into your branch, I added a formatting check to the CI, would like it to run on the PR to ensure the formatting is consistent

File: src/settings.ts [hunk 5]
```diff
@@ -5,6 +5,8 @@ export class MealSettings {
     meal_plan_note = 'Meal Plan';
     shopping_list_note = 'Shopping List';
 
+    recipe_format = 'RecipeMD';
```
Comment (tmayoff) [inline src/settings.ts:8]: Could this be an enum, and maybe defaulted to not use recipemd

**Review by tmayoff — CHANGES_REQUESTED** [review]

File: test_vault/.obsidian/plugins/obsidian-meals/manifest.json (hunk 4 above)
Comment (careilly) [reply test_vault/.obsidian/plugins/obsidian-meals/manifest.json:4]: That’s the current version, surely? Not a change I made anyway?

File: src/recipe/ingredients.ts (hunk 1 above)
Comment (careilly) [reply src/recipe/ingredients.ts:60]: I’ll take it out.

File: test_vault/.obsidian/plugins/obsidian-meals/manifest.json (hunk 4 above)
Comment (tmayoff) [reply test_vault/.obsidian/plugins/obsidian-meals/manifest.json:4]: oh this is in the test vault, I think this file was updated when you started the extension locally, I made the version bump in github online so it didn't update the this file. 

You can ignore this then

File: src/settings.ts (hunk 5 above)
Comment (careilly) [reply src/settings.ts]: Changed the default, will read up on JS enums.

**Comment by tmayoff** [comment]
Apologies for the conflicts, but you can remove .obsidian/... files and merge the .md files however you wish

**Comment by tmayoff** [comment]
I don't want to keep adding things for you to do, but obsidian added [this](https://github.com/obsidianmd/obsidian-api/blob/master/CHANGELOG.md#new-getfrontmatterinfo-utility), to their api for getting info where the frontmatter is. This is what I was wanting originally, making sure the ingredient parsing works whether or not there's frontmatter. 

Let me know if I've added too many things to your plate for this, and we can just merge this as is.

**Comment by careilly** [comment]
Of course they did. I'll have a look over weekend.

File: src/store.ts [hunk 6]
```diff
@@ -24,12 +24,13 @@ export const ingredients = derived(recipes, ($recipes) => {
     return ingredients;
 });
 
-export async function load_recipes() {
+export async function load_recipes(file:TFolder) {
```
Comment (tmayoff) [inline src/store.ts:27]: I believe the actual type here is a TAbstractFile

https://docs.obsidian.md/Reference/TypeScript+API/Vault/on.

Some pseudocode for making this a bit more intuitive. 

```
if file is folder and folder != recipe_folder return
if file is file and file.parent != recipe_folder return
```

**Review by tmayoff — APPROVED** [review]

**Comment by tmayoff** [comment]
Didn't mean to approve this, if you could address the review please

File: src/settings.ts (hunk 5 above)
Comment (careilly) [reply src/settings.ts]: Changed to use enums. Assuming I've used them correctly.

**Comment by tmayoff** [comment]
Unless there's anything else you wanted to add I'll merge this, just let me know

**Comment by careilly** [comment]
> Unless there's anything else you wanted to add I'll merge this, just let me know

No, I'm done for the moment: I'll use it for a while and see how it fits my use case.

**Comment by tmayoff** [comment]
Open a new PR if there's any more issues, thanks for the contribution!

**Pull request merged**